client/state/reader/sites/reducer.js

```diff
@@ -36,7 +36,8 @@ function adaptSite( attributes ) {
 		attributes.domain = withoutHttp( attributes.URL );
 		attributes.slug = attributes.domain.replace( /\//g, '::' );
 	}
-	attributes.title = trim( attributes.name ) || attributes.domain;
+	attributes.title =
+		( typeof attributes.name === 'string' && attributes.name.trim() ) || attributes.domain;
```

Comment:
Could use optional chaining with the same result: `attrs.name?.trim() || attrs.domain`.

Reply:
I wonder if `name` can be other than `string` or undefined. That being said, what Jarda suggests makes sense.

I also think it makes sense to do `?? attributes.domain` instead of `|| attributes.domain`.

Reply:
We don't have type safety that `.name` is a string which is why I added that guard here. The `||` is necessary with the current expression but could be changed to `??` if we switch to what Jarda suggested.